graph-proxy/src/graphql/workflows.rs

```diff
@@ -228,6 +234,38 @@ impl TryFrom<String> for TaskStatus {
     }
 }
 
+/// A task artifact
+#[derive(Debug, Clone, SimpleObject)]
+struct Artifact {
+    /// The name of the artifact
+    name: String,
+    /// The download URL for the artifact
+    url: Url,
+}
+
+impl Artifact {
+    /// Create a new `Artifact` from `IoArgoprojWorkflowV1alpha1Artifact` and associated metadata
+    fn new(
+        artifact: &IoArgoprojWorkflowV1alpha1Artifact,
+        visit: &str,
+        name: &str,
+        mut server_url: Url,
+    ) -> Artifact {
+        let artifact_name = &artifact.name[..];
+        server_url.path_segments_mut().unwrap().extend([
+            "artifacts",
+            visit,
+            name,
+            name,
```

Comment:
Surely this should be the `nodeId`? Which presumably comes from the task?

Reply:
Yeah that's right, one of the `name`'s is being used as the `nodeId` part in the endpoint, which is:
```
/artifacts/{namespace}/{name}/{nodeId}/{artifactName}
```

I noticed that the `nodeId` was always the same as the workflow name, so I just reused the workflow name for both parts in the path.

This was mainly because `IoArgoprojWorkflowV1alpha1Artifact` doesn't contain the `nodeId`, which is instead in `IoArgoprojWorkflowV1alpha1NodeStatus`, so I opted to be lazy and resuse in the `name` parameter.

Could grab the `nodeId` from the node status object and pass that through in addition to `name`, to make it less confusing (I do agree that the double use of `name` is confusing - and maybe it's incorrect to assume that the workflow name is always the same as the node ID).

Reply:
I would only do it this way if they were guaranteed to be the same, but somehow I doubt it's the case if they're exposed as different values